WordPress/Classes/ViewRelated/Blog/TimezoneSelectorViewModel.swift

```diff
@@ -0,0 +1,203 @@
+                if let highlightedIndexPath = indexPathToHighlight,
+                    highlightedIndexPath.section == sectionIndex {
+                    for (rowIndex, timezoneInfo) in allTimezones.enumerated() {
+                        let isHighlighted: Bool = rowIndex == highlightedIndexPath.row
+                        rows.append(TimezoneListRow(timezoneLabel: timezoneInfo.0, timezoneValue: timezoneInfo.1, action: action, highlighted: isHighlighted))
+                    }
+                } else {
+                    rows = allTimezones.map({ TimezoneListRow(timezoneLabel: $0.0, timezoneValue: $0.1, action: action, highlighted: false) })
+                }
+                let section = ImmuTableSection(headerText: continent, rows: rows)
+                sections.append(section)
+            }
+            return ImmuTable(sections: sections)
+        }
+    }
+
+    // MARK: Helper methods
+
+    /// Save API data to core data DB
+    func insertDataToDB(resultsDict: [String: [String: String]]) {
+        let manager = ContextManager.sharedInstance()
+        let context = manager.newDerivedContext()
+        context.performAndWait {
+            for (continent, timezonesInContinent) in resultsDict {
+                for (key, val) in timezonesInContinent {
+                    let timezoneInfo = NSEntityDescription.insertNewObject(forEntityName: "TimezoneInfo", into: context) as! TimezoneInfo
+                    timezoneInfo.label = key
+                    timezoneInfo.value = val
+                    timezoneInfo.continent = continent
+                }
+            }
+            manager.saveContextAndWait(context)
+        }
+    }
+
+    /// Helper method to fetch all TimezoneInfo objects from DB
+    func loadDataFromDB() -> [TimezoneInfo] {
+        let fetchRequest = NSFetchRequest<TimezoneInfo>(entityName: "TimezoneInfo")
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "continent", ascending: true),
+                                        NSSortDescriptor(key: "label", ascending: true)]
+        do {
+            return try ContextManager.sharedInstance().mainContext.fetch(fetchRequest)
+        } catch {
+            print(error)
+            return []
+        }
+    }
+
+    /// Returns a sorted list of continents and timezones mapped to a continent
+    private func setupVariables(with result: [TimezoneInfo]) -> ContinentsAndTimezones {
+        var continentNames = result.reduce(into: Set<String>(), { (mySet, timezone) in
+            mySet.insert(timezone.continent)
+        }).sorted()
+
+        // remove manual offset's from it's sorted position and push it to the bottom
+        if let manualOffsetIndex = continentNames.index(of: self.MANUAL_OFFSET) {
+            continentNames.remove(at: manualOffsetIndex)
+            continentNames.append(self.MANUAL_OFFSET)
+        }
+        var timezoneNamesSortedByContinent: [String: [(String, String)]] = [:]
```

Comment:
As mentioned above, a simple struct to wrap some of this data might help with readability here.